fdbrpc/Locality.cpp

```diff
@@ -148,6 +148,30 @@ ProcessClass::Fitness ProcessClass::machineClassFitness( ClusterRole role ) cons
 		default:
 			return ProcessClass::WorstFit;
 		}
+	case ProcessClass::DataDistributor:
+		// TODO: understand all the fitnesses and choose from them.
+		switch( _class ) {
+			case ProcessClass::DataDistributorClass:
+				return ProcessClass::BestFit;
+			case ProcessClass::StatelessClass:
+				return ProcessClass::GoodFit;
+			case ProcessClass::MasterClass:
+				return ProcessClass::OkayFit;
+			case ProcessClass::ResolutionClass:
+				return ProcessClass::OkayFit;
+			case ProcessClass::TransactionClass:
+				return ProcessClass::OkayFit;
+			case ProcessClass::ProxyClass:
+				return ProcessClass::OkayFit;
+			case ProcessClass::LogRouterClass:
+				return ProcessClass::OkayFit;
```

Comment:
Log routers will only ever be recruited in remote DCs, so I think you'd want to avoid placing data distribution here if you can.  I'd suggest removing this and letting it fall down into WorstFit.

Reply:
Thanks for pointing it out!